### ISSUE TITLE
False positive in rule explicit_type_interface with allowRedundancy

### ISSUE THREAD

**Issue by Dschee** [realm/SwiftLint#2636]
For this code

```swift
private let l10n = L10n.Communication.self
```

using this configuration

```yaml
explicit_type_interface:
  allow_redundancy: true
  excluded:
    - local
```

there shouldn't be any warnings, but there is.

Rationale: The usage of this rule here is mostly for improving readability, not performance of the compiler. We also try to prevent redundant type annotations, why explicitly writing the type here wouldn't be right. Thus this rules behavior must be changed.

**Comment by rocklyve** [comment]
with this example: 
```
static var `default` = Settings()
```
there is the warning `explicit_type_interface`, but there shouldn't be any warning. 
I think the backticks ` causes this warning.

Can you please fix this?
Thanks :)

**Comment by Dschee** [comment]
@rocklyve This issue was closed long ago. Please create a new one.
